File: app/src/main/java/com/weathercock/android_cleanarchitecture/presentation/di/module/DataModule.java
package com.weathercock.android_cleanarchitecture.presentation.di.module;

import android.content.Context;

import com.weathercock.android_cleanarchitecture.data.repository.AboutRepositoryImpl;
import com.weathercock.android_cleanarchitecture.data.repository.CityRepositoryImpl;
import com.weathercock.android_cleanarchitecture.data.repository.WeatherRepositoryImpl;
import com.weathercock.android_cleanarchitecture.domain.repository.AboutRepository;
import com.weathercock.android_cleanarchitecture.domain.repository.CityRepository;
import com.weathercock.android_cleanarchitecture.domain.repository.WeatherRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    @Provides
    @Singleton
    public CityRepository provideCityRepository(Context context) {
        return new CityRepositoryImpl(context);
    }

    @Provides
    @Singleton
    public WeatherRepository provideWeatherRepository(Context context) {
        return new WeatherRepositoryImpl(context);
    }

    @Provides
    @Singleton
    public AboutRepository provideAboutRepository(Context context) {
        return new AboutRepositoryImpl(context);
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/di/module/DataModule.kt
package com.kazakago.cleanarchitecture.di.module

import android.content.Context
import com.kazakago.cleanarchitecture.data.repository.AboutRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.CityRepositoryImpl
import com.kazakago.cleanarchitecture.data.repository.WeatherRepositoryImpl
import com.kazakago.cleanarchitecture.domain.repository.AboutRepository
import com.kazakago.cleanarchitecture.domain.repository.CityRepository
import com.kazakago.cleanarchitecture.domain.repository.WeatherRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class DataModule {

//    @Singleton
//    @Provides
//    fun provideOrmaDatabase(context: Context): OrmaDatabase {
//        return OrmaDatabase.builder(context)
//                .build()
//    }

    @Provides
    @Singleton
    fun provideCityRepository(context: Context): CityRepository {
        return CityRepositoryImpl(context)
    }

    @Provides
    @Singleton
    fun provideWeatherRepository(): WeatherRepository {
        return WeatherRepositoryImpl()
    }

    @Provides
    @Singleton
    fun provideAboutRepository(context: Context): AboutRepository {
        return AboutRepositoryImpl(context)
    }

}